tests/clv/models/test_pareto_nbd.py

```diff
@@ -336,67 +294,72 @@ def test_expected_purchase_probability(self, test_n, test_t):
+                {
+                    "r": [self.r_true],
+                    "alpha": [self.alpha_true],
+                    "s": [self.s_true],
+                    "beta": [self.beta_true],
+                }
+            )
+            self.model.idata = mock_fit
 
-        # TODO: A test with more than 1 chain and draw is required for full coverage.
-        chains = 2
-        draws = 50
-        mock_fit = az.from_dict(
-            {
-                "r": self.rng.normal(self.r_true, 1e-3, size=(chains, draws)),
-                "alpha": self.rng.normal(self.alpha_true, 1e-3, size=(chains, draws)),
-                "s": self.rng.normal(self.s_true, 1e-3, size=(chains, draws)),
-                "beta": self.rng.normal(self.beta_true, 1e-3, size=(chains, draws)),
-            }
-        )
-        rtol = 0.17
+        rtol = 0.26
 
-        mock_fit = az.from_dict(
-            {
-                "r": [self.r_true],
-                "alpha": [self.alpha_true],
-                "s": [self.s_true],
-                "beta": [self.beta_true],
-            }
+        customer_dropout = self.model.distribution_new_customer_dropout(
+            random_seed=self.rng
+        )
+        customer_purchase_rate = self.model.distribution_new_customer_purchase_rate(
+            random_seed=self.rng
         )
-        model.idata = mock_fit
 
-        rng = np.random.default_rng(42)
-        customer_dropout = model.distribution_new_customer_dropout(random_seed=rng)
-        customer_purchase_rate = model.distribution_new_customer_purchase_rate(
-            random_seed=rng
+        customer_rec_freq = self.model.distribution_customer_population(
+            random_seed=self.rng
         )
 
         assert isinstance(customer_dropout, xarray.DataArray)
         assert isinstance(customer_purchase_rate, xarray.DataArray)
+        assert isinstance(customer_rec_freq, xarray.DataArray)
 
         N = 1000
         lam = pm.Gamma.dist(alpha=self.r_true, beta=1 / self.alpha_true, size=N)
         mu = pm.Gamma.dist(alpha=self.s_true, beta=1 / self.beta_true, size=N)
-        rtol = 0.11
+
+        pop = ParetoNBD.dist(
+            r=self.r_true,
+            alpha=self.alpha_true,
+            s=self.s_true,
+            beta=self.beta_true,
+            T=self.T,
+        )
 
         np.testing.assert_allclose(
             customer_purchase_rate.mean(),
-            pm.draw(lam.mean(), random_seed=rng),
+            pm.draw(lam.mean(), random_seed=self.rng),
             rtol=rtol,
         )
         np.testing.assert_allclose(
             customer_purchase_rate.var(),
-            pm.draw(lam.var(), random_seed=rng),
+            pm.draw(lam.var(), random_seed=self.rng),
             rtol=rtol,
         )
         np.testing.assert_allclose(
-            customer_dropout.mean(), pm.draw(mu.mean(), random_seed=rng), rtol=rtol
+            customer_dropout.mean(), pm.draw(mu.mean(), random_seed=self.rng), rtol=rtol
```

Comment:
Having a `self.rng` makes the test depend on the order/quantity in which they are run which is not optimal. Each test should create an rng, or `self.rng` should be recreated for each test.